### PR TITLE
Kdh suggestions

### PR DESCRIPTION
**Pull request by kdharris101**
Here are some suggested changes to the Data models, to Timeseries, Electrophysiology, and Imaging.

### CODE REVIEW

**Comment by rossant** [comment]
Thanks! I'll have a look. Just a small suggestion: don't hesitate to delete lines of code instead of commenting them. It's easier to see what changed in the diff. There's no risk of losing things with git because deleted lines can always be recovered later.

**Comment by kdharris101** [comment]
Ok thanks, will do that in future.

What is the best way to communicate why I have deleted them? Just add one line of comment where they used to be?

**Comment by rossant** [comment]
I'd suggest to put comments not in the code but on GitHub. Click on "files changed" at the top of this pull request. Then scroll down to some line. You should see a blue "+" appear on the left. Click there and add some comment.

**Comment by kdharris101** [comment]
But how do you do that as you edit? I have been editing in a text editor on my machine, would I then need to commit and push every time I want to add a comment?

**Comment by rossant** [comment]
You could make all of your changes, propose your pull request, and then write all of your comments in one go on GitHub. If you really prefer to write comments while you are making changes in your text editor, then just write them as comments.

File: alyx/data/models.py [hunk 1]
```diff
@@ -4,26 +4,19 @@
...
 
-class DataRepository(BasePolymorphicModel):
+class DataRepository(BaseModel):
     """
     Base class for a file storage device; this could be a local hard-drive on a laptop,
     a network file location, or an offline archive tape / Blu-Ray disc / hard-drive.
     """
     name = models.CharField(max_length=255)
-    # TODO: type introspection for location type
+	type = models.CharField(max_length=255, help_text="e.g. web, local_disk, torrent")
```
Comment (rossant) [inline alyx/data/models.py]: * rename type to repository_type
* change to foreign key

File: alyx/behavior/models.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ class EventSeries(BaseExperimentalData):
     """
     event_times = models.ForeignKey(Dataset, blank=True, null=True,
                                     related_name="event_series_event_times",
-                                    help_text="n*1 array of times in seconds")
+                                    help_text="n*1 array of times in seconds (universal timescale)")
     type_descriptions_id = models.ForeignKey(Dataset, blank=True, null=True)
     event_types_id = models.ForeignKey(Dataset, blank=True, null=True,
                                        related_name="event_series_event_descriptions",
```
Comment (rossant) [inline alyx/behavior/models.py]: IntervalSeries

* rename interval_description to interval_types
* add type_descriptions

File: alyx/electrophysiology/models.py [hunk 3]
```diff
@@ -25,27 +25,29 @@ def classname(self):
...
                                       help_text="Extracellular high-passed data")
     filter_info = models.CharField(max_length=255, blank=True,
                                    help_text="Details of hardware corner frequencies, filter "
                                    "type, order. TODO: make this more structured?")
-    start_time = models.FloatField(null=True, blank=True,
-                                   help_text="in seconds relative to session start.")
+    # KDH: not sure why we need start_time at all! it is taken care of by Timeseries fields!
+	#
+	#start_time = models.FloatField(null=True, blank=True,
+    #                               help_text="in seconds relative to session start.")
```
Comment (rossant) [inline alyx/electrophysiology/models.py]: keep but rename to nominal_start_time
same with end_time

File: alyx/electrophysiology/models.py [hunk 4]
```diff
@@ -101,18 +103,18 @@ class SpikeSorting(BaseExperimentalData):
...
+    unfiltered_waveforms = models.ForeignKey(Timeseries, blank=True, null=True,
                                                  related_name="spike_sorting_"
                                                  "unfiltered_waveforms",
                                                  help_text="mean unfiltered waveforms of "
                                                  "every spike on every channel")
-    mean_filtered_waveform = models.ForeignKey(Dataset, blank=True, null=True,
+    filtered_waveforms = models.ForeignKey(Timeseries, blank=True, null=True,
                                                related_name="spike_sorting_filtered_waveforms",
                                                help_text="mean filtered waveforms of every spike "
                                                "on every channel")
```
Comment (rossant) [inline alyx/electrophysiology/models.py]: should be mean waveform, move to SpikeSortedUnit instead

File: alyx/imaging/models.py [hunk 5]
```diff
@@ -6,24 +6,28 @@
...
+    raw_data = models.ForeignKey(Timeseries, blank=True, null=True, related_name="widefield_raw",
                                  help_text="pointer to nT by nX by nY by nC (colors) binary file")
-    compressed_data = models.ForeignKey('SVDCompressedMovie', null=True, blank=True,
+    compressed_data = models.ForeignKey(SVDCompressedMovie, null=True, blank=True,
                                         related_name="widefield_compressed",
                                         help_text="Link to SVD compressed movie, "
                                         "if compression was run")
-    start_time = models.FloatField(null=True, blank=True,
-                                   help_text="in seconds relative to session start. "
-                                   "TODO: not DateTimeField? / TimeDifference")
```
Comment (rossant) [inline alyx/imaging/models.py]: * rename to nominal_start_time and same for end time
* DateTimeField

File: alyx/imaging/models.py [hunk 6]
```diff
@@ -6,24 +6,28 @@
...
-    imaging_indicator = models.CharField(max_length=255, blank=True,
-                                         help_text="<GCaMP6f, GCaMP6m, GCaMP6s, "
-                                         "VSFPb1.2, intrinsic, …>. TODO: normalize!")
+    #end_time = models.FloatField(null=True, blank=True,
+    #                             help_text="Equals start time if single application. "
+    #                             "TODO: should this be an offset? Or DateTimeField? "
+    #                             "Or TimeDifference?")
+	
+	# KDH: not necessary since genotype/virus has this information
+    #imaging_indicator = models.CharField(max_length=255, blank=True,
```
Comment (rossant) [inline alyx/imaging/models.py]: go back

File: alyx/electrophysiology/models.py (hunk 4 above)
Comment (rossant) [inline alyx/electrophysiology/models.py]: get rid of generating software and provenance directory

File: alyx/electrophysiology/models.py [hunk 7]
```diff
@@ -101,18 +103,18 @@ class SpikeSorting(BaseExperimentalData):
     spike_times = models.ForeignKey(Dataset, blank=True, null=True,
```
Comment (rossant) [inline alyx/electrophysiology/models.py]: extracellular_recording is a ForeignKey

**Comment by kdharris101** [comment]
Changes to timeseries models

**Pull request closed without merging**